### ISSUE TITLE
keep session open for several command

### ISSUE THREAD

**Issue by lujiacn** [senseyeio/roger#14]
Hi Daniel, 

Now, in roger struct, defined <code> address *net.TCPAddr</code>, there is no session (or rconnection). Each Eval function will be closed and the variables evaluated will be released.
It make the code below not possible
<code>
r.Eval("x <- 1")
</code>
<code>
x, err := r.Eval(x) // since it is a new connection, x will not exit in R.</br>
</code>

Do you think we can define <code> sess *session</code> in roger struct
And each r.Eval will use the same r.sess.
Of course define a r.Close() to close the connection when no r scripts will be evaluated. 

Regards,
Jia

**Comment by lujiacn** [comment]
And Daniel,
I found VoidEval is not necessary. The issue before cased "unknown error" may be due to "\r" in r_script.
And I removed voidEval functions. Before pull any requests, just confirm with you. 
And also move the *session to roger struct. All updated in my forked repo

**Comment by dareid** [comment]
Yea I agree we need to support the reuse of a session in some instances. However I think the most common use case would be a new session per command, this is critical for putting R behind a web api. 
Maybe we could have a new client constructor which creates a reusable session within the Roger struct. For the existing client constructors the session in the Roger struct would be nil and a new session started per command. 
What do you think?

**Comment by dareid** [comment]
Could you create a pull request for the removal of voideval, then we can have a play with how best to support session reuse.

**Comment by lujiacn** [comment]
Yes, I will clean the code and submit a pull request. Thanks :)

**Comment by lujiacn** [comment]
And the session re-use, I will also think about it in details.

**Comment by dareid** [comment]
Just looking at the code. Maybe we make a new Session interface which is implemented by the current session struct. From the client you can request a Session, which you can then run multiple commands on and close afterwards. I think the Session interface could just expose the current `SendCommand` and `Close` methods.

**Comment by lujiacn** [comment]
instead of make another Session interface, I combined *session to roger. Due to limited programming skills.
My assumption about the connect and communication processes are:
-----------------------
r := NewRconnection(host, user, password) // the rclient connection keep alive
r.Eval("any R script")  // execute single or R block script
variable, err := r.Eval("x") // get "x" variable value in r script
output, err := r.Eval("long r script.. ") // output will be the last variable in the r script
r.Close()
-----------------------
I saw the session structure defined connected and will be marked as false when session closed. Not sure how to use it.

And EvalSync and Evaluate, will be single connection for each command. Seems the Evalute function should accept  commands []string, not command string. The multiple rscripts can be evaluated sync.

I wish the questions will not interrupt your programming rhythm  :)
